Microsoft.TeamServices.Samples.Client/WorkItemTracking/QueriesSample.cs

```diff
@@ -12,24 +13,401 @@ public class QueriesSample : ClientSample
+
+                Console.WriteLine("Query Successfully Created");
+                Console.WriteLine("Id:         {0}", query.Id);
+                Console.WriteLine("Name:       {0}", query.Name);
+                Console.WriteLine("Path:       {0}", query.Path);
+
                 return query;
             }
+            catch (System.AggregateException ex)
+            {
+                if (ex.InnerException.Message.Contains("TF237018"))
+                {
+                    Console.ForegroundColor = ConsoleColor.Yellow;
+                    Console.WriteLine("Error creating query: Query name in specified path already exists");
+                    Console.ForegroundColor = ConsoleColor.White;
+                }
+
+                return null;
+            }         
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem CreateFolder()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryPath = "Shared Queries";
+
+            QueryHierarchyItem postedQuery = new QueryHierarchyItem()
+            {
+                Name = "Folder " + System.Guid.NewGuid().ToString(), //random folder name
+                IsFolder = true
+            };
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            QueryHierarchyItem query = workItemTrackingClient.CreateQueryAsync(postedQuery, projectId, queryPath).Result;
+
+            if (query == null)
+            {
+                Console.WriteLine("Error creating folder");
+            }
+            else
+            {
+                Console.WriteLine("Folder Successfully Created");
+                Console.WriteLine("Id:         {0}", query.Id);
+                Console.WriteLine("Name:       {0}", query.Name);
+                Console.WriteLine("Path:       {0}", query.Path);
+            }
+
+            return query;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem UpdateQuery()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            string queryId = "2614c4de-be48-4735-9fdc-9656f55c495f";
+
+            QueryHierarchyItem queryUpdate = new QueryHierarchyItem()
+            {               
+                Wiql = "Select [System.Id], [System.Title], [System.State] From WorkItems Where [System.WorkItemType] = 'Bug' AND [System.State] = 'Active' order by [Microsoft.VSTS.Common.Priority] asc, [System.CreatedDate] desc"
+            };
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            QueryHierarchyItem query = workItemTrackingClient.UpdateQueryAsync(queryUpdate, projectId, queryId).Result;
+
+            if (query == null)
```

Comment:
I'd expect this method to throw if there was an error (like if the user didn't have permission to update the specified query).